src/main/java/com/gregtechceu/gtceu/api/recipe/OverclockingLogic.java

```diff
@@ -33,62 +41,56 @@ public interface OverclockingLogic {
-
-    default @NotNull ModifierFunction getModifierNoParallel(@NotNull MetaMachine machine, @NotNull GTRecipe recipe,
-                                                            long maxOverclockVoltage) {
-        long EUt = Math.abs(RecipeHelper.getRealEUt(recipe));
-        if (EUt > 0) {
-            return performOverclocking(EUt, recipe.duration, maxOverclockVoltage, 1).asModifierFunction();
-        }
-        return ModifierFunction.IDENTITY;
+        if (subtick) return (params, maxV) -> subTickParallelOC(params, maxV, durationFactor, voltageFactor);
+        else return (params, maxV) -> standardOC(params, maxV, durationFactor, voltageFactor);
     }
 
     /**
-     * Calculates the overclocked Recipe's final duration and EU/t
-     *
-     * @param recipe the recipe to run
-     * @return a new recipe
+     * Determines overclocking parameters from the given arguments, runs the overclock, and returns a ModifierFunction
+     * 
+     * @param machine        machine
+     * @param recipe         recipe
+     * @param maxVoltage     max overclock voltage
+     * @param shouldParallel whether the OC Logic should parallel or not
+     * @return A {@link ModifierFunction} describing how the OC application should modify the recipe
      */
-    default @NotNull ModifierFunction getModifier(@NotNull MetaMachine machine, @NotNull GTRecipe recipe,
-                                                  long maxOverclockVoltage) {
+    default @NotNull ModifierFunction getModifier(MetaMachine machine, GTRecipe recipe,
+                                                  long maxVoltage, boolean shouldParallel) {
         long EUt = Math.abs(RecipeHelper.getRealEUt(recipe));
-        if (EUt > 0) {
-            int recipeTier = GTUtil.getTierByVoltage(EUt);
-            int maximumTier = GTUtil.getTierByVoltage(maxOverclockVoltage);
-
-            int maxParallels;
-            if (this == PERFECT_OVERCLOCK || this == NON_PERFECT_OVERCLOCK) { // short-circuit for these common OCs
-                maxParallels = 1;
-            } else if ((Math.pow(PERFECT_DURATION_FACTOR, maximumTier - recipeTier) * recipe.duration) > 1) {
-                maxParallels = Integer.MAX_VALUE; // if duration cannot possibly go below 1, give it max parallels
-            } else {
-                maxParallels = ParallelLogic.getParallelAmount(machine, recipe, Integer.MAX_VALUE);
-            }
+        if (EUt == 0) return ModifierFunction.IDENTITY;
 
-            return performOverclocking(EUt, recipe.duration, maxOverclockVoltage, maxParallels).asModifierFunction();
+        int recipeTier = GTUtil.getTierByVoltage(EUt);
+        int maximumTier = GTUtil.getTierByVoltage(maxVoltage);
+        int OCs = maximumTier - recipeTier;
+        if (recipeTier == GTValues.ULV) OCs--;
+        if (OCs == 0) return ModifierFunction.IDENTITY;
+
+        int maxParallels;
+        if (!shouldParallel || this == PERFECT_OVERCLOCK || this == NON_PERFECT_OVERCLOCK) { // don't parallel
```

Comment:
It's just a short circuit so no math has to be done, since those common instances of oc logic are guaranteed not to subtick. Just helps performance as to not accidentally do the parallel logic call.

Reply:
well, it already checks `!shouldParallel`, that's why I assumed they have no need to exist